coordinator/schedulepod/main.go

```diff
@@ -0,0 +1,694 @@
+}
+
+var testOfflineTableValues = [...]provider.ResourceRecord{
+	provider.ResourceRecord{Entity: "a", Value: 1, TS: time.UnixMilli(0).UTC()},
+	provider.ResourceRecord{Entity: "b", Value: 2, TS: time.UnixMilli(0).UTC()},
+	provider.ResourceRecord{Entity: "c", Value: 3, TS: time.UnixMilli(0).UTC()},
+	provider.ResourceRecord{Entity: "d", Value: 4, TS: time.UnixMilli(0).UTC()},
+	provider.ResourceRecord{Entity: "e", Value: 5, TS: time.UnixMilli(0).UTC()},
+}
+
+var postgresConfig = provider.PostgresConfig{
+	Host:     "postgres",
+	Port:     "5432",
+	Database: "postgresdb",
+	Username: "postgresadmin",
+	Password: "admin123",
+	// Database: os.Getenv("POSTGRES_DB"),
+	// Username: os.Getenv("POSTGRES_USER"),
+	// Password: os.Getenv("POSTGRES_PASSWORD"),
+}
+
+// postgresPassword: "StrongPassword"
+// username: "app1"
+// password: "AppPassword"
+// database: "app_db"
+
+// var redisPort = os.Getenv("REDIS_PORT")
+var redisPort = "6379"
+var redisHost = "redis"
+
+var etcdHost = "localhost"
+var etcdPort = "2379"
+
+func startServ() (*metadata.MetadataServer, string) {
+	logger := zap.NewExample().Sugar()
+	storageProvider := metadata.EtcdStorageProvider{
+		metadata.EtcdConfig{
+			Nodes: []metadata.EtcdNode{
+				{etcdHost, etcdPort},
+			},
+		},
+	}
+	config := &metadata.Config{
+		Logger:          logger,
+		StorageProvider: storageProvider,
+	}
+	serv, err := metadata.NewMetadataServer(config)
+	if err != nil {
+		panic(err)
+	}
+	// listen on a random port
+	lis, err := net.Listen("tcp", ":0")
+	if err != nil {
+		panic(err)
+	}
+	go func() {
+		if err := serv.ServeOnListener(lis); err != nil {
+			panic(err)
+		}
+	}()
+	return serv, lis.Addr().String()
+}
+
+func createNewCoordinator(addr string) (*coordinator.Coordinator, error) {
+	logger := zap.NewExample().Sugar()
+	client, err := metadata.NewClient(addr, logger)
+	if err != nil {
+		return nil, err
+	}
+	etcdConnect := fmt.Sprintf("%s:%s", etcdHost, etcdPort)
+	cli, err := clientv3.New(clientv3.Config{Endpoints: []string{etcdConnect}})
+	if err != nil {
+		return nil, err
+	}
+	kubeJobSpawner := coordinator.KubernetesJobSpawner{}
+	return coordinator.NewCoordinator(client, logger, cli, &kubeJobSpawner)
+}
+
+func testScheduleTrainingSet() error {
+	if err := runner.RegisterFactory(string(runner.CREATE_TRAINING_SET), runner.TrainingSetRunnerFactory); err != nil {
+		return fmt.Errorf("Failed to register training set runner factory: %v", err)
+	}
+	defer runner.UnregisterFactory(string(runner.CREATE_TRAINING_SET))
+	logger := zap.NewExample().Sugar()
+	client, err := metadata.NewClient(metadata_addr, logger)
+	if err != nil {
+		return fmt.Errorf("Failed to connect: %v", err)
+	}
+	defer client.Close()
+	etcdConnect := fmt.Sprintf("%s:%s", etcdHost, etcdPort)
+	cli, err := clientv3.New(clientv3.Config{Endpoints: []string{etcdConnect}})
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+	featureName := createSafeUUID()
+	labelName := createSafeUUID()
+	tsName := createSafeUUID()
```

Comment:
What is tsName for?

Reply:
name of the training set for the ts resource